coap-core/src/main/java/com/mbed/coap/transport/TransportContext.java

```diff
@@ -66,6 +68,17 @@ public <T> TransportContext with(Key<T> key, T value) {
         return new TransportContext(requireNonNull(key), requireNonNull(value), this);
     }
 
+    public TransportContext with(TransportContext otherCtx) {
```

Comment:
Maybe we should add some javadoc to explain that :
If a key exists in both contexts, the `otherCtx` value will be used.